util/src/test/java/tk/zielony/randomdata/util/StringIPAddressGeneratorTest.java

```diff
@@ -0,0 +1,31 @@
+package tk.zielony.randomdata.util;
+
+import android.util.Log;
+
+import org.junit.Test;
+
+import static junit.framework.Assert.assertEquals;
+
+public class StringIPAddressGeneratorTest {
+
+    @Test
+    public void generationIsCorrect() throws Exception {
+        StringIPAddressGenerator ipAddressGenerator = new StringIPAddressGenerator();
+        for (StringIPAddressGenerator.classRange range : StringIPAddressGenerator.classRange.values()) {
+            ipAddressGenerator.setClassRange(range);
+            String ipAddress = ipAddressGenerator.nextInternal(null);
```

Comment:
tests should use Generator.next() instead of Generator.nextInternal(). I should probably make Generator.nextInternal() protected so it's only used by subclasses what was my initial intention